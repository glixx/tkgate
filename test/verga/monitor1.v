module top;
  reg [7:0] a,o,x,b,no,nx,na;

  assign a = & b;
  assign o = | b;
  assign x = ^ b;

  assign na = ~& b;
  assign no = ~| b;
  assign nx = ~^ b;

  initial $monitor("%2h:  &:%2h  |:%2h  ^:%2h  ~&:%2h  ~|:%2h  ~^:%2h",b,a,o,x,na,no,nx);

  initial
    begin
	#1 b = 8'h0;
	#1 b = 8'h4;
	#1 b = 8'h7;
	#1 b = 8'hff;
	#1 b = 8'hfz;
	#1 b = 8'h8z;
	#1 b = 8'h0z;
    end

endmodule
