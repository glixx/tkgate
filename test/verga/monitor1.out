 0:  &: 0  |: 0  ^: 0  ~&: 1  ~|: 1  ~^: 1
 4:  &: 0  |: 1  ^: 1  ~&: 1  ~|: 0  ~^: 0
 7:  &: 0  |: 1  ^: 1  ~&: 1  ~|: 0  ~^: 0
ff:  &: 1  |: 1  ^: 0  ~&: 0  ~|: 0  ~^: 1
fz:  &: x  |: 1  ^: x  ~&: x  ~|: 0  ~^: x
8z:  &: 0  |: 1  ^: x  ~&: 1  ~|: 0  ~^: x
0z:  &: 0  |: x  ^: x  ~&: 1  ~|: x  ~^: x
